From 0617858997f26c8e64759468dd7ebf3886ba44f2 Mon Sep 17 00:00:00 2001
From: Sergii Piatakov <sergii.piatakov@globallogic.com>
Date: Thu, 15 Nov 2018 15:21:34 +0200
Subject: [PATCH 2/2] improve calculation accuracy

Allow using float point arguments to avoid truncation.

Test: Add (4.9, 4.9) should return 10.
Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>

fix truncation error

To convert float to integer the truncation is performed, but the
rounding is expected.

Test: Add (4.9, 4.9) should return 10.
Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
Signed-off-by: Maryna Moroziuk <marynam60@gmail.com>

formatting: use tabs instead of spaces

Signed-off-by: Sergii Piatakov <sergii.piatakov@globallogic.com>
---
 calculator.cpp | 16 ++++++++++++----
 calculator.h   |  6 +++---
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/calculator.cpp b/calculator.cpp
index 3c63184..639822a 100644
--- a/calculator.cpp
+++ b/calculator.cpp
@@ -1,11 +1,19 @@
 #include "calculator.h"
 
-int Calculator::Add (int a, int b)
+int Calculator::Add (double a, double b)
 {
-    return a + b;
+
+
+
+
+	return a + b + 0.5;
+
+
+
+
 }
 
-int Calculator::Sub (int a, int b)
+int Calculator::Sub (double a, double b)
 {
-    return Add (a, -b);
+	return Add (a, -b);
 }
diff --git a/calculator.h b/calculator.h
index 84ff414..d59d596 100644
--- a/calculator.h
+++ b/calculator.h
@@ -3,9 +3,9 @@
 
 class Calculator
 {
-    public:
-        int Add (int, int);
-        int Sub (int, int);
+	public:
+		int Add (double, double);
+		int Sub (double, double);
 };
 
 #endif//CALCULATOR_H
-- 
2.47.1.windows.1

